catboost/private/libs/hyperparameter_tuning/hyperparameter_tuning.cpp

```diff
@@ -920,171 +920,174 @@ namespace {
         TProfileInfo profile(gridIterator->GetTotalElementsCount());
         while (auto paramsSet = gridIterator->Next()) {
             profile.StartIterationBlock();
-            // paramsSet: {border_count, feature_border_type, nan_mode, [others]}
-            TQuantizationParamsInfo quantizationParamsSet;
-            quantizationParamsSet.BinsCount = GetRandomValueIfNeeded((*paramsSet)[0], randDistGenerators).GetInteger();
-            quantizationParamsSet.BorderType = FromString<EBorderSelectionType>((*paramsSet)[1].GetString());
-            quantizationParamsSet.NanMode = FromString<ENanMode>((*paramsSet)[2].GetString());
-
-            AssignOptionsToJson(
-                TConstArrayRef<TString>(paramNames),
-                TConstArrayRef<NJson::TJsonValue>(
-                    paramsSet->begin() + IndexOfFirstTrainingParameter,
-                    paramsSet->end()
-                ), // Ignoring quantization params
-                randDistGenerators,
-                modelParamsToBeTried
-            );
+            try {
```

Comment:
It isn't ok:)
Let's create a function IsValidParams and check the options for validity in it.
After that write the following code here:
```
if (!IsValidParams(paramsSet)) {
    continue;
}
```